output/compact_activity.go

```diff
@@ -7,21 +7,20 @@ import (
 	"github.com/pganalyze/collector/output/transform"
 	"github.com/pganalyze/collector/state"
 	"github.com/pganalyze/collector/util"
-	pg_query "github.com/pganalyze/pg_query_go/v4"
 )
 
 func SubmitCompactActivitySnapshot(ctx context.Context, server *state.Server, grant state.Grant, collectionOpts state.CollectionOpts, logger *util.Logger, activityState state.TransientActivityState) error {
 	as, r := transform.ActivityStateToCompactActivitySnapshot(server, activityState)
 
-	if server.Config.FilterQuerySample == "all" {
+	if server.Config.FilterQueryText != "none" {
 		for idx, backend := range as.Backends {
 			// Normalize can be slow, protect against edge cases here by checking for cancellations
```

Comment:
Does this mean that we will likely have more collectors going through this potentially slow part? (as `FilterQuerySample == "all"` is somewhat "special case" (aka not default), whereas `FilterQueryText != "none"` is actually default. Is it okay change to make?

Reply:
Thats a fair point, and I could indeed see that being a problem on collectors with many servers configured. I wonder if we should maybe not make this the default for now (since query samples are also not filtered by default), and just keep using the filter_query_sample setting here, but also do this normalization if its set to "normalize" (not just "all").

Reply:
Adjust this to continue working based on `filter_query_sample` for now, but to also do this when its set to `normalize` (not just `all`).